plugins/UFPWriter/UFPWriter.py

```diff
@@ -212,19 +213,28 @@ def _getObjectMetadata(node: SceneNode) -> List[Dict[str, str]]:
                 if item.getMeshData() is not None and not item.callDecoration("isNonPrintingMesh")]
 
     def _getSettings(self) -> Dict[str, Dict[str, Dict[str, str]]]:
-        container_registry = Application.getInstance().getContainerRegistry()
-
-        global_stack = cast(GlobalStack, Application.getInstance().getGlobalContainerStack())
-        quality_changes = global_stack.qualityChanges
-
         settings = {
-            "extruder_1": {
-                "changes": {},
-                "default": {}
-            },
-            "extruder_2": {
+            "global": {
                 "changes": {},
                 "default": {}
-            },
+            }
         }
+
+        global_stack = cast(GlobalStack, Application.getInstance().getGlobalContainerStack())
+
+        for i, extruder in enumerate(global_stack.extruderList):
+            extruder_flattened_changes = CuraStackBuilder.createFlattenedContainerInstance(extruder.userChanges, extruder.qualityChanges)
+            settings[f"extruder_{i}"] = {}
+            settings[f"extruder_{i}"]["changes"] = {}
+            settings[f"extruder_{i}"]["default"] = {}
```

Comment:
can't these lines be combined in a single expression?
```suggestion
            settings[f"extruder_{i}"] = {
                ["changes"]: {},
                ["default"]: {},
            }
```